### PR TITLE
CDD-2091: Review logging in api-notification-pull for incoming requests

### PR DESCRIPTION
**Pull request by rashadmughal**

### CODE REVIEW

File: test/unit/fakes/SuccessfulHeaderValidatorFake.scala [hunk 1]
```diff
@@ -16,12 +16,14 @@
 
 package unit.fakes
 
+import org.scalatest.mockito.MockitoSugar
 import play.api.mvc.{ActionBuilder, Request, Result}
+import uk.gov.hmrc.apinotificationpull.logging.NotificationLogger
 import uk.gov.hmrc.apinotificationpull.validators.HeaderValidator
 
 import scala.concurrent.Future
```
Comment (googley42) [inline test/unit/fakes/SuccessfulHeaderValidatorFake.scala]: `with MockitoSugar` not required

File: app/uk/gov/hmrc/apinotificationpull/connectors/ApiNotificationQueueConnector.scala [hunk 2]
```diff
@@ -19,32 +19,44 @@ package uk.gov.hmrc.apinotificationpull.connectors
...
   def getById(notificationId: String)(implicit hc: HeaderCarrier): Future[Option[Notification]] = {
-    http.GET[HttpResponse](s"$serviceBaseUrl/notification/$notificationId")
+
+    val url = s"$serviceBaseUrl/notification/$notificationId"
+    logger.debug(s"Getting notification by id using url: $url", hc.headers)
+    http.GET[HttpResponse](url)
       .map { r =>
+        logger.debug(s"Notification received successfully with id: $notificationId", hc.headers)
         Some(Notification(notificationId, r.allHeaders.map(h => h._1 -> h._2.head), r.body))
       }
```
Comment (googley42) [inline app/uk/gov/hmrc/apinotificationpull/connectors/ApiNotificationQueueConnector.scala]: should be
```scala
      .recover[Option[Notification]] {
        case _: NotFoundException => {
          logger.debug(s"Notification not found with id: $notificationId", hc.headers)
          None
        }
      }
```

File: app/uk/gov/hmrc/apinotificationpull/connectors/EnhancedApiNotificationQueueConnector.scala [hunk 3]
```diff
@@ -20,23 +20,29 @@ import javax.inject.Inject
...
 
   def getNotificationBy(notificationId: String, notificationStatus: NotificationStatus.Value)(implicit hc: HeaderCarrier): Future[Either[HttpException, Notification]] = {
 
-    http.GET[HttpResponse](s"$serviceBaseUrl/notifications/${notificationStatus.toString}/$notificationId")
+    val url = s"$serviceBaseUrl/notifications/${notificationStatus.toString}/$notificationId"
+    logger.debug(s"Calling get notifications by using url: $url", hc.headers)
+    http.GET[HttpResponse](url)
       .map { r =>
         Right(Notification(notificationId, r.allHeaders.map(h => h._1 -> h._2.head), r.body))
       }
```
Comment (googley42) [inline app/uk/gov/hmrc/apinotificationpull/connectors/EnhancedApiNotificationQueueConnector.scala]: should be
```scala
      .recover[Either[HttpException, Notification]] {
        case nfe: NotFoundException => Left(nfe)
        case bre: BadRequestException => Left(bre)
        case ise => Left(new InternalServerException(ise.getMessage))
      }
```

File: app/uk/gov/hmrc/apinotificationpull/logging/NotificationLogger.scala [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+import com.google.inject.Inject
+import javax.inject.Singleton
+import uk.gov.hmrc.apinotificationpull.logging.LoggingHelper.formatWithHeaders
+import uk.gov.hmrc.apinotificationpull.model.SeqOfHeader
+import uk.gov.hmrc.customs.api.common.logging.CdsLogger
+
+@Singleton
+class NotificationLogger @Inject()(logger: CdsLogger) {
+
```
Comment (googley42) [inline app/uk/gov/hmrc/apinotificationpull/logging/NotificationLogger.scala]: I would consider making 2nd parameter `headers: => SeqOfHeader` implicit.
You could also create an implicit function or class that takes a HeaderCarrier as input and returns a `headers: => SeqOfHeader` . This should help to cut down on some of the repetition

File: test/unit/util/CustomHeaderNames.scala [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+ * limitations under the License.
+ */
+
+package unit.util
+
+object CustomHeaderNames {
+
+  val X_CLIENT_ID_HEADER_NAME = "X-Client-ID"
+
+}
```
Comment (paulblakehmrc) [inline test/unit/util/CustomHeaderNames.scala]: perhaps a little ott having its own file - consider maybe a `RequestHeaders` object as part of `TestData`?

Comment (rashadmughal) [reply test/unit/util/CustomHeaderNames.scala]: I'd like to keep it consistent with our other microservices, even if it's only one header.

File: test/unit/logging/NotificationLoggerSpec.scala [hunk 6]
```diff
@@ -31,19 +35,26 @@ class NotificationLoggerSpec extends UnitSpec with MockitoSugar {
...
     val logger = new NotificationLogger(mockCdsLogger)
+
+    implicit val mockHeaderCarrier: HeaderCarrier = mock[HeaderCarrier]
+    when(mockHeaderCarrier.headers).thenReturn(LoggingHeaders)
   }
 
   "NotificationsLogger" should {
     "debug(s: => String, headers: => SeqOfHeader)" in new SetUp {
-      logger.debug("msg", LoggingHeaders)
+      import uk.gov.hmrc.apinotificationpull.controllers.CustomHeaderNames.getHeadersFromHeaderCarrier
```
Comment (googley42) [inline test/unit/logging/NotificationLoggerSpec.scala]: signature of `debug(s: => String, headers: => SeqOfHeader)` has changed and should be reflected here if you want to keep the same format. Same applies to other tests below

File: test/unit/util/CustomHeaderNames.scala (hunk 5 above)
Comment (paulblakehmrc) [reply test/unit/util/CustomHeaderNames.scala]: Do some of our services have a `CustomHeaderNames` in test code?

**Pull request merged**